### DISCUSSION TITLE
Why acknowledge-mode:auto is unsafe?

### DISCUSSION THREAD

**Discussion by nopromise** [spring-projects/spring-amqp#2549] [Q&A] [answered]
If acknowledge-mode are configured auto,I find SimpleMessageListenerContainer will use channel to ack when the message are processed completely.I don't know why autoAck is considered unsafe,because if the message are not processed completely,SimpleMessageListenerContainer will not ack and the message is still in queue.

**Comment by garyrussell** [comment, marked as answer]
Where do you see it being described as unsafe?

Perhaps you are confused with Spring's `AcknowledgeMode.AUTO` Vs. RabbitMQ's `autoAck`.

The names are unfortunate but, with Spring, RabbitMQ's `autoAck` is configured using `AcknowledgeMode.NONE`.

WIth Spring, AUTO means the container "automatically" acks the message after successful processing. NONE means the broker acks the message after it is sent to the consumer.

**Reply by nopromise** [reply]
Thank you for your reply.I did confuse Spring's `AcknowledgeMode.AUTO` with RabbitMQ's `autoAck`.
